### PR TITLE
Specify scope of default environment variables

### LINKED ISSUE #24895: default environment variables - where can they be used?
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

### What part(s) of the article would you like to see updated?

the section on default environment variables. it says

> The default environment variables that GitHub sets are available to every step in a workflow.

but when I try for instance to use such a variable as input to a step using a composite action, as in

```
      - name: Fetch job html
        uses: ./.github/actions/job_html_url
        with:
          job_name: ${{ env.GITHUB_JOB }}
```

the passed input is the empty string.

I would like to see more specifically where in a step these env vars can be used, as i would expect that they can be accessed using the `env` context in expressions, but apparently this is not the case. not even a step with
```
        run: echo ${{ env.GITHUB_JOB }}
```
works! it seems they are not accessible through the env context at all :thinking: 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
It is not obvious that default environment variables will not be available in the `env` context. We add a note to that regard.

### Why:

Closes: #24895 

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@valentin-krasontovitsch Thank you for opening a PR and linking it to your issue! ✨ 

I'll get this triaged for review! :zap:

**Comment by valentin-krasontovitsch** [comment]
four months for a simple, small docs update. I have my own project that I'm woefully neglecting, so I understand this can happen, but that one has just one maintainer, and is free-time based :sweat_smile:

**Comment by cmwilson21** [comment]
@valentin-krasontovitsch Thanks for checking in on this. It is up on the board for review 👍 

We appreciate your patience as we are working through our backlog 💖

**Comment by valentin-krasontovitsch** [comment]
Of course, thank you for your response, and godspeed :rainbow:

**Comment by SiaraMist** [comment]
Hey @valentin-krasontovitsch, thank you so much for your patience with us as we work through our backlog. 🙇‍♀️ 

I do see how it would be confusing for users to know how to use default environment variables in workflows with the way the docs are currently written. There is [a note](https://github.com/github/docs/pull/24896/files#diff-e0e008dd4d4b5d933f0ed768ad05251d30e5da99be5b5a1af0f33e72aef52d51R344) underneath the table of default environment variables that says:

> Most of the default variables have a corresponding, and similarly named, context property. For example, the value of the `GITHUB_REF` variable can be read during workflow processing using the `${{ github.ref }}` context property.

What do you think of this idea:

We could move that out of the note and into its own paragraph after the first sentence in the introduction. We could also add a sentence before it so that we say something along the lines of:

> Because default environment variables are set by GitHub and not defined in a workflow, they are not accessed through the `env` context. However, most of the default variables have a corresponding, and similarly named, context property. For example, the value of the `GITHUB_REF` variable can be read during workflow processing using the `${{ github.ref }}` context property.

Would that make things more clear in this doc? Let me know! Thank you for helping to improve GitHub's docs! ⚡

**Comment by valentin-krasontovitsch** [comment]
yes i think that would definitely help! thanks for the much better suggestion : )

should i change this PR to reflect your suggestions, or close this one and leave it to you / somebody else to create a new PR?

**Comment by SiaraMist** [comment]
That's awesome. 😃 @valentin-krasontovitsch if you want to update this PR, then I can accept your changes and get them merged!

**Comment by valentin-krasontovitsch** [comment]
it is done ^^

let me know if the new diff is OK, and if the commit message is up to par.

File: content/actions/learn-github-actions/variables.md
Comment (SiaraMist) [inline content/actions/learn-github-actions/variables.md:345]:
```suggestion
**Note:** If you need to use a workflow run's URL from within a job, you can combine these variables: `$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID`
```

File: content/actions/learn-github-actions/variables.md
Comment (SiaraMist) [inline content/actions/learn-github-actions/variables.md:277]:
```suggestion
Because default environment variables are set by {% data variables.product.prodname_dotcom %} and not defined in a workflow, they are not accessible through the `env` context. However, most of the default variables have a corresponding, and similarly named, context property. For example, the value of the `GITHUB_REF` variable can be read during workflow processing using the {% raw %}`${{ github.ref }}`{% endraw %} context property.
```
Just adding some formatting!

**Review by SiaraMist — COMMENTED** [review]
Your updates look great, as does your commit message! ⚡ I just added some docs formatting, and if it lets me commit my suggestions I'll merge this now.

**Review by SiaraMist — APPROVED** [review]

**Pull request merged**
